src/streamlink/stream/hls.py

```diff
@@ -72,53 +83,83 @@ def create_decryptor(self, key, sequence):
             self.key_data = res.content
             self.key_uri = key_uri
 
-        iv = key.iv or num_to_iv(sequence)
+        iv = key.iv or self.num_to_iv(num)
 
         # Pad IV if needed
         iv = b"\x00" * (16 - len(iv)) + iv
 
         return AES.new(self.key_data, AES.MODE_CBC, iv)
 
-    def create_request_params(self, sequence):
+    def create_request_params(self, segment: Union[Segment, Map]):
         request_params = dict(self.reader.request_params)
         headers = request_params.pop("headers", {})
 
-        if sequence.segment.byterange:
-            bytes_start = self.byterange_offsets[sequence.segment.uri]
-            if sequence.segment.byterange.offset is not None:
-                bytes_start = sequence.segment.byterange.offset
+        if segment.byterange:
+            bytes_start = self.byterange_offsets[segment.uri]
+            if segment.byterange.offset is not None:
+                bytes_start = segment.byterange.offset
 
-            bytes_len = max(sequence.segment.byterange.range - 1, 0)
+            bytes_len = max(segment.byterange.range - 1, 0)
             bytes_end = bytes_start + bytes_len
-            headers["Range"] = "bytes={0}-{1}".format(bytes_start, bytes_end)
-            self.byterange_offsets[sequence.segment.uri] = bytes_end + 1
+            headers["Range"] = f"bytes={0}-{1}".format(bytes_start, bytes_end)
```

Comment:
`.format` can be removed

Reply:
Good catch! That was a undo/redo mistake which I didn't notice while staging/committing my changes.